### PR TITLE
[COST-4904] Update aws resource matching to include csi_volume_handle

### PR DESCRIPTION
**Pull request by myersCody**
## Jira Ticket

[COST-4904](https://issues.redhat.com/browse/COST-4904)

## Description

This change will update the aws resource matching to utilize the csi_volume_handle

## Testing

1. Checkout Branch
2. Run load test customer data for AWS
3. Run this trino query:
```
select lineitem_resourceid, resource_id_matched  from aws_openshift_daily where lineitem_resourceid like 'vol-%' group by lineitem_resourceid, resource_id_matched;
```
Results:
```
 lineitem_resourceid | resource_id_matched
---------------------+---------------------
 vol-12345672        | true
 vol-12345673        | true
 vol-12345671        | true
 vol-12345674        | true
 ```
 ^ This used to return false

## Release Notes
- [ ] proposed release note

```markdown
* [COST-####](https://issues.redhat.com/browse/COST-####) Fix some things
```

### CODE REVIEW

**Review by lcouzens — DISMISSED** [review]

**Comment by myersCody** [comment]
So, there is an additional edge case to consider with these changes. Now the operator is already release, so some customer may have already upgraded. That means that these resources will have resource_id_matched value of True, which excludes them from the tag matching logic [here](https://github.com/project-koku/koku/blob/main/koku/masu/database/trino_sql/reporting_ocpawscostlineitem_daily_summary.sql#L337).

In order to maintain the same logic until I get the unattributed storage pieces delivered, we will need to still consider the csi volume handle resources for tag matching. Otherwise cost correlation could drop unexpectedly when we roll these changes out.

**Comment by myersCody** [comment]
```
trino:org1234567> select distinct matched_tag, resource_id from aws_openshift_daily_tag_matched_temp where resource_id like 'vol-%';
        matched_tag        | resource_id
---------------------------+--------------
 "storageclass": "delta"   | vol-12345672
 "storageclass": "epsilon" | vol-12345673
 ```

File: koku/masu/database/trino_sql/reporting_ocpawscostlineitem_daily_summary.sql [hunk 1]
```diff
@@ -347,6 +347,114 @@ GROUP BY aws.lineitem_usagestartdate,
...
+        THEN array_union(ARRAY['openshift_cluster', 'openshift_node', 'openshift_project'], array_agg(key))
+        ELSE ARRAY['openshift_cluster', 'openshift_node', 'openshift_project']
+    END as enabled_keys
+    FROM postgres.{{schema | sqlsafe}}.reporting_enabledtagkeys
+    WHERE enabled = TRUE
+    AND provider_type = 'AWS'
+),
+cte_csi_volume_handles as (
+    SELECT distinct csi_volume_handle as csi_volume_handle
+            FROM hive.org1234567.openshift_storage_usage_line_items_daily as ocp
```
Comment (lcouzens) [inline koku/masu/database/trino_sql/reporting_ocpawscostlineitem_daily_summary.sql:389]: I'm pretty sure this is wrong :teehee:

**Review by lcouzens — CHANGES_REQUESTED** [review]

File: koku/masu/database/trino_sql/reporting_ocpawscostlineitem_daily_summary.sql (hunk 1 above)
Comment (myersCody) [reply koku/masu/database/trino_sql/reporting_ocpawscostlineitem_daily_summary.sql:389]: good catch

File: koku/masu/database/trino_sql/reporting_ocpawscostlineitem_daily_summary.sql [hunk 2]
```diff
@@ -347,6 +347,114 @@ GROUP BY aws.lineitem_usagestartdate,
...
+    END as product_code,
+    nullif(aws.product_productfamily, '') as product_family,
+    nullif(aws.product_instancetype, '') as instance_type,
+    max(aws.lineitem_usageaccountid) as usage_account_id,
+    nullif(aws.lineitem_availabilityzone, '') as availability_zone,
+    nullif(aws.product_region, '') as region,
+    max(nullif(aws.pricing_unit, '')) as unit,
+    sum(aws.lineitem_usageamount) as usage_amount,
+    max(nullif(aws.lineitem_currencycode, '')) as currency_code,
+    sum(aws.lineitem_unblendedcost) as unblended_cost,
```
Comment (lcouzens) [inline koku/masu/database/trino_sql/reporting_ocpawscostlineitem_daily_summary.sql:409]: You will probably merge your PR first but as a side note and reminder to myself I guess. We will need to update the unblended/blended inserts in here for negation costs. Following this https://github.com/project-koku/koku/pull/5252

**Review by lcouzens — APPROVED** [review]
Left one comment, which tbh I can deal with afterwards. So looks good to me.

**Pull request merged**